libtelemetry/src/main/java/com/mapbox/android/telemetry/TelemetryClient.java

```diff
@@ -37,6 +40,18 @@
   private TelemetryClientSettings setting;
   private final Logger logger;
   private CertificateBlacklist certificateBlacklist;
+  private MetricUtils metricUtils;
```

Comment:
should we re-consider overloading TelemetryClient class with new fields and adhere to object oriented design principles? IMHO TelemetryClient is exposed to way too many implementation details of counters logic when it shouldn't. I'd rather see member class that hides all this complexity and exposes very simple API to the TelemetryClient class.